internal/api/handlers/url.go

```diff
@@ -0,0 +1,71 @@
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/GlebRadaev/shlink/internal/config"
+	"github.com/GlebRadaev/shlink/internal/service"
+	"github.com/go-chi/chi/v5"
+)
+
+// URLHandlers defines the handlers for URL shortening
+type URLHandlers struct {
+	urlService *service.URLService
+	config     *config.Config
+}
+
+// NewURLHandlers creates a new instance of URLHandlers
+func NewURLHandlers(urlService *service.URLService, config *config.Config) *URLHandlers {
+	return &URLHandlers{
+		urlService: urlService,
+		config:     config,
+	}
+}
+
+// Shorten handles the request to shorten a URL
+func (h *URLHandlers) Shorten(w http.ResponseWriter, r *http.Request) {
+	contentType := r.Header.Get("Content-Type")
+	parts := strings.Split(contentType, ";")
+	if parts[0] != "text/plain" {
```

Comment:
[CHANGE REQUEST] This code is too dangerous.
https://pkg.go.dev/strings#Contains